### DISCUSSION TITLE
Change local folder

### DISCUSSION THREAD

**Discussion by rickshory** [community/community#43447] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

How do I change the folder of my local repository, and keep it connected to the same online repository, with history preserved? I explored 'git mv', but it gave the error that the destination folder was outside the current repository. I need to move the repository files to a completely different folder.

**Comment by antoninpicard** [comment]
Crée un nouveau dossier où tu veux déplacer ton référentiel local.
Ouvre un terminal dans le nouveau dossier et utilise la commande git clone pour cloner ton dépôt en ligne dans ce dossier. Par exemple :
Copy code
git clone https://github.com/ton_nom_dutilisateur/ton_depot.git
Copie tous les fichiers de ton référentiel local dans le nouveau dossier du référentiel cloné.
Ouvre un terminal dans le nouveau dossier du référentiel et utilise la commande git add pour ajouter tous les fichiers copiés au prochain commit :
Copy code
git add .
Utilise la commande git commit pour enregistrer les changements dans l'historique du dépôt :
Copy code
git commit -m "Déplacement du référentiel local dans un nouveau dossier"
Enfin, utilise la commande git push pour envoyer les commits au dépôt en ligne :
Copy code
git push origin master
Cela devrait déplacer ton référentiel local vers le nouveau dossier tout en conservant son lien avec le dépôt en ligne et en préservant l'historique des commits.
